core/store.go

```diff
@@ -60,7 +64,7 @@ func NewObj(value interface{}, expDurationMs int64, oType uint8, oEnc uint8) *Ob
 	return obj
 }
 
-func Put(k string, obj *Obj) {
+func Put(k string, obj *Obj, options *PutOptions) {
```

Comment:
Many calls to the `Put` API don't seem to be using any options, additionally, it may be more flexible to use a functional pattern here.

```go
func Put(k string, obj *Obj, opts ...PutOption) {
   // This could set the TTL option to false by default.
    options := defaultPutOptions()
    for _, opt := range opts {
        opt(options)
    }

    // ... rest of the function
}

// type of function that takes a PutOptions pointer as input.
type PutOption func(*PutOptions)

func WithKeepTTL() PutOption {
    return func(o *PutOptions) {
        o.KeepTTL = true
    }
}

// Usage:
Put("key", obj, WithKeepTTL())
```
A big benefit of this approach is that you could choose to not pass any options at all to the `Put` API, reducing the number of changes you need to do in this PR.

Reply:
@soumya-codes what do you think about this suggestion?